### PR TITLE
Fix api info for File component

### PR DESCRIPTION
**Pull request by freddyaboulton**
# Description

There were two issues before:
- The type agnostic info for `File(file_count='multiple')` showed that it could be either an array or a single item. Now it only shows it can be an array
- Example inputs for `File(file_count='multiple')` will now show a list. This will only apply for new demos going forward.

image

In order for CI to pass we need to release a new version of the python client

# Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have added a short summary of my change to the CHANGELOG.md
- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

# A note about the CHANGELOG

Hello 👋 and thank you for contributing to Gradio!

All pull requests must update the change log located in CHANGELOG.md, unless the pull request is labeled with the "no-changelog-update" label.

Please add a brief summary of the change to the Upcoming Release > Full Changelog section of the CHANGELOG.md file and include
a link to the PR (formatted in markdown) and a link to your github profile (if you like). For example, "* Added a cool new feature by `[@myusername](link-to-your-github-profile)` in `[PR 11111](https://github.com/gradio-app/gradio/pull/11111)`".

If you would like to elaborate on your change further, feel free to include a longer explanation in the other sections.
If you would like an image/gif/video showcasing your feature, it may be best to edit the CHANGELOG file using the 
GitHub web UI since that lets you upload files directly via drag-and-drop.

### CODE REVIEW

File: client/python/gradio_client/utils.py
Comment (freddyaboulton) [inline client/python/gradio_client/utils.py:32]: Will install this PR to the v2 api fetcher space once it's approved

File: gradio/blocks.py
Comment (freddyaboulton) [inline gradio/blocks.py:554]: This is the same logic applied to inputs also applied to outputs. I think it was just an oversight it wasn't done before.

**Review by pngwn — APPROVED** [review]
Thanks @freddyaboulton! Looks good to me but I'll let someone else take a look as well. I'll also need to switch the version check in the js client like you have here.

**Comment by abidlabs** [comment]
I'm confused why we have this special case:

```py
                if (
                    isinstance(serializer, serializing.FileSerializable)
                    and component["props"].get("file_count", "single") != "single"
                ):
                    python_info = serializer._multiple_file_serialized_info()
```

in `get_api_info()` in `blocks.py`. Shouldn't it pull from `FileSerializable` since the same logic exists there?

Nit:
* The `FileSerializable` class is starting to have a lot of one-line methods, whereas it would be a lot simpler just to have if/else statements. For example, we can replace:

```py
    def example_inputs(self) -> dict[str, Any]:
        return self._single_file_example_inputs()

    def _single_file_example_inputs(self) -> dict[str, Any]:
        return {
            "raw": {"is_file": False, "data": media_data.BASE64_FILE},
            "serialized": "https://github.com/gradio-app/gradio/raw/main/test/test_files/sample_file.pdf",
        }

    def _multiple_file_example_inputs(self) -> dict[str, Any]:
        return {
            "raw": [{"is_file": False, "data": media_data.BASE64_FILE}],
            "serialized": [
                "https://github.com/gradio-app/gradio/raw/main/test/test_files/sample_file.pdf"
            ],
        }
```

with 

```py
    def example_inputs(self, file_count="single") -> dict[str, Any]:
        if file_count=="single":
          return {
              "raw": {"is_file": False, "data": media_data.BASE64_FILE},
              "serialized": "https://github.com/gradio-app/gradio/raw/main/test/test_files/sample_file.pdf",
          }
        else:
          return {
              "raw": [{"is_file": False, "data": media_data.BASE64_FILE}],
              "serialized": [
                  "https://github.com/gradio-app/gradio/raw/main/test/test_files/sample_file.pdf"
              ],
          }
```
which is more condensed

**Comment by abidlabs** [comment]
The actual funcitonality looks good though, LGTM

**Review by abidlabs — APPROVED** [review]

**Review by abidlabs — APPROVED** [review]

**Comment by freddyaboulton** [comment]
Thanks for the review @abidlabs ! 

We pull the serialized data format by instantiating the serializer but that serializer doesn't know about single vs multiple since that's from the component. So we have to check the component config to see if it's a multiple file component and then get the multiple file info from the serializer.

Yea I get what you mean about the single line methods but I don't think it's good to add a parameter to example_inputs as the api for example_inputs should be the same across serializers.

**Comment by abidlabs** [comment]
I see, thanks for the explanation. The info generation is starting to get a bit messy and hard to maintain IMO. Let’s tackle that in a future PR. This LGTM

**Comment by freddyaboulton** [comment]
Yes! I think a lot of the complexity comes from having to return the serialized types in the server. In an ideal world, the serialized types are an implementation detail of the client and the server wouldn't know about that.

**Pull request merged**
